### ISSUE TITLE
Map element access inside for loop causes: Floating point exception

### ISSUE THREAD

**Issue by jordalgo** [bpftrace/bpftrace#3142]
Trying to access map elements inside a `for` loop causes a "Floating point exception" if the map gets defined in a later probe. This  also seems to happen using the `-dd` option before it prints generated LLVMIR. Not yet sure why this is happening but below are two simple repros.

```
$ sudo bpftrace -e 'kretfunc:handle_mm_fault { for ($kv : @mp) { delete(@mp[$kv.0]); } } END { @mp[1] = 2; }'
Floating point exception
$ sudo bpftrace -e 'kretfunc:handle_mm_fault { for ($kv : @mp) { print((@mp[$kv.0])); } } END { @mp[1] = 2; }'
Floating point exception
$ sudo bpftrace -e 'BEGIN { @mp[1] = 2; } kretfunc:handle_mm_fault { for ($kv : @mp) { delete(@mp[$kv.0]); } }'
Attaching 2 probes...
```

**Comment by jordalgo** [comment]
Actually, this seems to be fixed in trunk (maybe fixed here: https://github.com/bpftrace/bpftrace/pull/3116). The error is for an earlier version of bpftrace.

**Comment by ajor** [comment]
For the record, this was fixed in #3112
